app/models/activity.rb

```diff
@@ -81,6 +96,7 @@ class Activity < ApplicationRecord
   FORM_STATE_VALIDATION_LIST = FORM_STEPS.map(&:to_s).push("complete")
 
   before_validation :strip_control_characters_from_fields!
+  before_save :set_non_oda_defaults, if: %i[is_oda_changed? is_non_oda?]
```

Comment:
I'll probably change this to `if: -> { is_oda_changed? && is_non_oda? }`. I think that's clearer.